stout/flags.cc

```diff
@@ -72,6 +126,39 @@ void Parser::AddOrExit(
 
 ////////////////////////////////////////////////////////////////////////
 
+google::protobuf::Message* Parser::TryParseSubcommand(const std::string& arg) {
+  CHECK(cur_subcommand_message_);
+
+  if (const google::protobuf::FieldDescriptor* subcommand_field =
+          cur_subcommand_message_
+              .value()
+              ->GetDescriptor()
+              ->FindFieldByName(arg);
+      subcommand_field == nullptr) {
+    // There is no any field with the name 'arg'.
+    return nullptr;
+  } else {
+    if (const google::protobuf::OneofDescriptor* real_oneof =
+            subcommand_field->real_containing_oneof();
+        real_oneof == nullptr) {
+      return nullptr;
+    } else {
+      if (real_oneof->name() != "subcommand") {
+        return nullptr;
+      } else {
```

Comment:
Should this be a `CHECK()` because you've already checked it before? Or is there another case that you're thinking of where this can happen?

Reply:
In the `else` statement on the line 148 I return nested message pointer in order to use this pointer for the following arguments in the cmd line. There is no need to put CHECK here cause I do right check after the call `TryParseSubcommand`.